src/recommendation/src/test/scala/com/microsoft/ml/spark/RecommendationTestBase.scala

```diff
@@ -0,0 +1,689 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import java.util.Random
+
+import com.github.fommil.netlib.BLAS.{getInstance => blas}
+import org.apache.spark.ml.Pipeline
+import org.apache.spark.ml.feature.StringIndexer
+import org.apache.spark.ml.linalg.{Vectors => mlVectors}
+import org.apache.spark.ml.recommendation.ALS.Rating
+import org.apache.spark.ml.tuning.ParamGridBuilder
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import org.apache.spark.{SparkConf, SparkContext}
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+trait RecommendationTestBase extends TestBase {
+  private[spark] val conf: SparkConf = new SparkConf()
+    .setAppName("Testing Recommendation Model")
+    .setMaster("local[*]")
+    .set("spark.driver.memory", "60g")
+    .set("spark.driver.maxResultSize", "0")
+
+  override lazy val session: SparkSession = SparkSession.builder()
+    .config(conf)
+    .getOrCreate()
+  session.sparkContext.setLogLevel("WARN")
+
+  private[spark] val sc2 = session.sqlContext
+
+  override lazy val sc: SparkContext = session.sparkContext
+
+  private[spark] lazy val df = {
+    session.createDataFrame(Seq(
```

Comment:
I think it would be nicer if these were put in 3 separate files.  I think we can do that in a separate PR after this.